lib/chewy/type.rb

```diff
@@ -41,6 +41,18 @@ def self.type_name
       adapter.type_name
     end
 
+    # Returns the identifier value
+    #
+    def self.identify objects
+      if root_object.id
+        identifiers_from_evaluator = root_object.compose_id(objects)
+      else
+        identifiers_from_evaluator = nil
+      end
+
+      Array identifiers_from_evaluator || adapter.identify(objects)
```

Comment:
make it `Array(identifiers_from_evaluator || adapter.identify(objects))` please ;)